### PR TITLE
Sort connected dapps by most recent handshake

### PR DESCRIPTION
**Pull request by pugson**
Fixes RNBW-2070

## What changed (plus any additional context for devs)
- ~~Added a timestamp to show the latest handshake with dapp~~
- Changed list sorting to be by most recent handshake
- Fixed L2 icon alignment on Android
- Fixed wallet icon shadow clipping at the bottom of the list on Android
- Fixed margin mismatch between Android and iOS
- Fixed name truncating for long wallet and app names

## PoW (screenshots / screen recordings)
The order is properly sorted by most recent.

|  Before  |  After: Android  | After: iOS | After: iOS cont. |
|----------|------------------|------------|------------------|
| Screenshot_20220314-020831 | Screenshot_20220316-180017 | IMG_0768| IMG_0770 |

<details>
<summary>Older screenshots</summary>

|  Before  |  After  | After changes |
|----------|---------|---------------|
| Screenshot_20220314-020831 | Screenshot_20220314-021003 | Screenshot_20220316-004507 |
</details>

## Dev checklist for QA: what to test
Connecting wallets, switching networks, switching wallets, signing messages and transactions with connected wallets.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

**Review by osdnk — APPROVED** [review]
from the code perspective, this is a no-brainer. although I'm not.a fan of negative margins. However, the design review here is more relevant.

File: src/helpers/time.js [hunk 1]
```diff
@@ -57,3 +58,15 @@ export const getMinimalTimeUnitStringForMs = (
 
   return `${highestResolutionValue} ${label}`;
 };
+
+/**
+ * @desc get time in microseconds and return a timestamp string
+ * @param {String} [handshake='']
+ * @return {String}
+ */
```
Comment (osdnk) [inline src/helpers/time.js:66]: Appreciate this, but realistically this is not needed. TS should fix this eventually

**Comment by pugson** [comment]
oh wow yeah these timestamps should probably go

**Comment by pugson** [comment]
Newest preview after changes:
Screenshot_20220316-004507

**Review by tchayen — CHANGES_REQUESTED** [review]
Doesn't look scary on small phones anymore, but veryveryverylongnames still break layout a bit, even on iPhone 13:

photo_2022-03-16 17 18 14

**Comment by pugson** [comment]
Thanks @tchayen. Will clean this up even though it's out of scope for this ticket but very much an ugly problem.

**Comment by pugson** [comment]
Ready to merge as-is because there are some more layout issues out of scope. New task for them: https://linear.app/rainbow/issue/RNBW-2843/connected-apps-list-margins-and-alignment-messed-up-on-android

**Review by tchayen — APPROVED** [review]
LGTM :shipit: (pls fix linter)

**Pull request merged**